### PR TITLE
Support all HTTP methods and JSON payload

### PR DESCRIPTION
**Pull request by neomilium**
Before this commit, only GET method with an empty payload (ie. body) was supported due to the hardcoded strings in authentication hash computation.

This commit allows to use any supported HTTP method and computes hash with body too.

Please note that due to the usage of faraday here and the lack of methods in it to retrieve the effective URI with query string and the computed body after middlewares processes, we have to compute them ourselve and pry that will match.

In other words, using `JSON.generate(data)` and `URI.encode_www_form(params)` to compute body and URI respectively is working now but could fail if Faraday decides to change its JSON serialization or the query string encoding.

### CODE REVIEW

**Comment by neomilium** [comment]
@SebastianJ Hi, what can I do to help in order to have a new release with this patch?

**Comment by neomilium** [comment]
@SebastianJ up?

**Comment by neomilium** [comment]
@SebastianJ Hello, any help to maintain this?

**Comment by neomilium** [comment]
@SebastianJ Do not let it reach 1-year old stalled status ;-)
